datafusion/core/src/execution/session_state.rs

```diff
@@ -967,14 +967,19 @@ impl SessionState {
             let field_access_planner =
                 Arc::new(functions_array::planner::FieldAccessPlanner) as _;
 
-            query
+            query = query
                 .with_user_defined_planner(array_planner)
                 .with_user_defined_planner(field_access_planner)
         }
-        #[cfg(not(feature = "array_expressions"))]
+        #[cfg(feature = "unicode_expressions")]
         {
-            query
+            let position_planner =
+                Arc::new(functions::unicode::planner::PositionPlanner::default()) as _;
+
+            query = query.with_user_defined_planner(position_planner);
```

Comment:
As part of this issues #https://github.com/apache/datafusion/pull/11253 we have deided to move registration of SQL planner to constructor `pub fn new_with_config_rt_and_catalog_list`